meta/data.go

```diff
@@ -622,6 +622,11 @@ func (sgi *ShardGroupInfo) Contains(timestamp time.Time) bool {
 	return !sgi.StartTime.After(timestamp) && sgi.EndTime.After(timestamp)
 }
 
+// Overlaps return whether the shard group contains data for the time range between min and max
+func (sgi *ShardGroupInfo) Overlaps(min, max time.Time) bool {
+	return !sgi.StartTime.After(max) && sgi.EndTime.After(min)
+}
```

Comment:
This is expected behavior. The example you show SHOULD hit both shard groups because the end time is on the hour, which is in the next one. Here's what it looks like if you're only hitting a single shard: http://play.golang.org/p/ypATbwAJQB